### DISCUSSION TITLE
How to set attribute "dangerouslySetInnerHTML" with ellipsis using antd?

### DISCUSSION THREAD

**Discussion by Nikita279** [ant-design/ant-design#29459] [Q&A]
I need to show comment with **ellipsis**. I have used antd's Paragraph **Typography** for it. 
My problem is that comment can also contain html attributes (link to tagged users) so I also need to set **dangerouslySetInnerHTML** in the component. How to set this in Typography component?
```
<Paragraph ellipsis={{ rows: 2, expandable: true, symbol: "more" }}>
      {comment}
</Paragraph>
```
**Preview:**
<img width="318" alt="Screenshot 2021-02-24 at 5 13 52 PM" src="https://user-images.githubusercontent.com/40610477/108995984-dd64c080-76c3-11eb-8d10-baeeaf3dd218.png">

I tried using span inside Paragraph to use dangerouslySetInnerHTML, but then ellipsis started showing just "...more" for all long comments, without showing any initial characters in the comment to fill the width.

```
<Paragraph ellipsis={{ rows: 2, expandable: true, symbol: "more" }}>
      <span dangerouslySetInnerHTML={{ __html: comment.comment }} />
</Paragraph>
```
**Preview:**
<img width="320" alt="Screenshot 2021-02-24 at 5 14 25 PM" src="https://user-images.githubusercontent.com/40610477/108995938-c9b95a00-76c3-11eb-97cc-39c4152a3f5d.png">

Any workaround for achieving this??

**Comment by zelief** [comment]
Try using [html-react-parser](https://www.npmjs.com/package/html-react-parser)

```
import parse from 'html-react-parser';

...

<Paragraph ellipsis={{ rows: 2, expandable: true, symbol: "more" }}>
      {parse(comment)}
</Paragraph>

```
